### PR TITLE
Go to DNX

### PR DESCRIPTION
**Pull request by ilmax**
Changed projects to use dnx
Make nancyfx sample run on dnx
Removed Basic/MVC samples

### CODE REVIEW

File: samples/Nancy/Nancy.Server/App.config [hunk 1]
```diff
@@ -94,7 +99,7 @@
     <defaultConnectionFactory type="System.Data.Entity.Infrastructure.SqlConnectionFactory, EntityFramework">
       <parameters>
         <!-- Replace "mssqllocaldb" by "v11.0" if you don't use SQL Server 2014 -->
-        <parameter value="Data Source=(localdb)\mssqllocaldb; Integrated Security=True; MultipleActiveResultSets=True;AttachDbFilename=|DataDirectory|ApplicationContext.mdf" />
+        <parameter value="Data Source=(localdb)\mssqllocaldb; Integrated Security=True; MultipleActiveResultSets=True; Initial Catalog=Nancy.Server.Sample" />
```
Comment (ilmax) [inline samples/Nancy/Nancy.Server/App.config]: I need to undo this, but cannot make it work using DataDirectory

File: samples/Nancy/Nancy.Server/Modules/AuthorizationModule.cs [hunk 2]
```diff
@@ -44,9 +44,10 @@ public class AuthorizationModule : NancyModule {
                 // Note: authentication could be theorically enforced at the filter level via AuthorizeAttribute
                 // but this authorization endpoint accepts both GET and POST requests while the cookie middleware
                 // only uses 302 responses to redirect the user agent to the login page, making it incompatible with POST.
-                // To work around this limitation, the OpenID Connect request is automatically saved in the cache and will be
-                // restored in the other "Authorize" method, after the authentication process has been completed.
-                if (OwinContext.Authentication.User.Identity == null ||
+                // To work around this limitation, the OpenID Connect request is automatically saved in the user's session and will
+                // be restored in the other "Authorize" method, after the authentication process has been completed.
+                if (OwinContext.Authentication.User == null ||
```
Comment (ilmax) [inline samples/Nancy/Nancy.Server/Modules/AuthorizationModule.cs]: This may be removed, I was facing NRE when converted this to dnx

File: samples/Nancy/Nancy.Server/Modules/AuthorizationModule.cs [hunk 3]
```diff
@@ -67,12 +68,12 @@ public class AuthorizationModule : NancyModule {
...
 
                 // Note: in a real world application, you'd probably prefer creating a specific view model.
-                return View["Authorize.cshtml", Tuple.Create(request, application)];
+                return View["Authorize.cshtml", new { Request = request, Resources = request.GetResources() , UniqueId = request.GetUniqueIdentifier(), Application = application }];
             };
             
             Post["/connect/authorize/accept", runAsync: true] = async (parameters, cancellationToken) => {
                 this.RequiresMSOwinAuthentication();
-                this.ValidateCsrfToken();
+                //this.ValidateCsrfToken(); TODO Investigate failure and uncomment me
```
Comment (ilmax) [inline samples/Nancy/Nancy.Server/Modules/AuthorizationModule.cs]: Need to open an issue to investigate this?

File: samples/Nancy/Nancy.Server/Nancy.Server.Old.csproj [hunk 4]
```diff
@@ -181,16 +181,6 @@
     <Compile Include="Startup.cs" />
   </ItemGroup>
   <ItemGroup>
```
Comment (ilmax) [inline samples/Nancy/Nancy.Server/Nancy.Server.Old.csproj]: this whole file needs to be deleted

File: samples/Nancy/Nancy.Server/Views/Logout.cshtml [hunk 5]
```diff
@@ -2,7 +2,7 @@
 @using Nancy.ViewEngines.Razor
 @using Microsoft.Owin.Security
 
-@inherits NancyRazorViewBase<Tuple<OpenIdConnectMessage, AuthenticateResult>>
+@inherits NancyRazorViewBase<System.Tuple<OpenIdConnectMessage, AuthenticateResult>>
```
Comment (ilmax) [inline samples/Nancy/Nancy.Server/Views/Logout.cshtml]: System. seems unnecessary

File: src/Owin.Security.OpenIdConnect.Server/project.json [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  "projectUrl": "https://github.com/aspnet-contrib/AspNet.Security.OpenIdConnect.Server",
+  "licenseUrl": "http://www.apache.org/licenses/LICENSE-2.0.html",
+
+  "dependencies": {
+    "Owin": "1.0.0",
+    "Microsoft.Owin": "3.0.1",
+    "Microsoft.Owin.Security": "3.0.1",
+    "Newtonsoft.Json": "6.0.8",
+    "System.IdentityModel.Tokens.Jwt": "4.0.2.202250711",
+    "Owin.Security.OpenIdConnect.Extensions": "1.0.0-*"
```
Comment (ilmax) [inline src/Owin.Security.OpenIdConnect.Server/project.json]: need to change this to ensure it's consuming a project reference instead of a package

File: samples/Nancy/Nancy.Client/App.config [hunk 7]
```diff
@@ -25,7 +25,8 @@
   </system.diagnostics>
   
   <system.web>
-    <compilation debug="true" targetFramework="4.5"><buildProviders>
+    <compilation debug="true" targetFramework="4.5">
```
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Client/App.config]: Question: are `System.Web` references needed for self-hosting scenarios?

File: samples/Nancy/Nancy.Client/Startup.cs [hunk 8]
```diff
@@ -68,11 +69,11 @@ public class Startup {
 
                             var request = new HttpRequestMessage(HttpMethod.Post, configuration.TokenEndpoint);
                             request.Content = new FormUrlEncodedContent(new Dictionary<string, string> {
-                                { OpenIdConnectParameterNames.ClientId, notification.Options.ClientId },
-                                { OpenIdConnectParameterNames.ClientSecret, notification.Options.ClientSecret },
-                                { OpenIdConnectParameterNames.Code, notification.ProtocolMessage.Code },
-                                { OpenIdConnectParameterNames.GrantType, "authorization_code" },
-                                { OpenIdConnectParameterNames.RedirectUri, notification.Options.RedirectUri }
+                                {OpenIdConnectParameterNames.ClientId, notification.Options.ClientId},
```
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Client/Startup.cs]: Please consider reverting these changes :+1:

File: samples/Nancy/Nancy.Client/App.config (hunk 7 above)
Comment (ilmax) [inline samples/Nancy/Nancy.Client/App.config]: well, maybe not :) will have a look

File: samples/Nancy/Nancy.Client/Startup.cs (hunk 8 above)
Comment (ilmax) [inline samples/Nancy/Nancy.Client/Startup.cs]: will do

File: samples/Nancy/Nancy.Server/App.config [hunk 9]
```diff
@@ -62,6 +62,11 @@
         <assemblyIdentity name="System.IdentityModel.Tokens.Jwt" publicKeyToken="31bf3856ad364e35" culture="neutral" />
         <bindingRedirect oldVersion="0.0.0.0-4.0.2.28" newVersion="4.0.2.28" />
       </dependentAssembly>
+
+      <dependentAssembly>
```
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Server/App.config]: Do you need these binding redirects when running under DNX?

File: samples/Nancy/Nancy.Server/Modules/AuthorizationModule.cs (hunk 3 above)
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Server/Modules/AuthorizationModule.cs]: I'll clone your fork and give it a try! :+1:

File: samples/Nancy/Nancy.Client/App.config [hunk 10]
```diff
@@ -0,0 +1,45 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<!--
+  For more information on how to configure your ASP.NET application, please visit
+  http://go.microsoft.com/fwlink/?LinkId=169433
+  -->
+<configuration>
+  <configSections>
+    <section name="razor" type="Nancy.ViewEngines.Razor.RazorConfigurationSection, Nancy.ViewEngines.Razor" />
+    
+    <sectionGroup name="system.web.webPages.razor" type="System.Web.WebPages.Razor.Configuration.RazorWebSectionGroup, System.Web.WebPages.Razor, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31BF3856AD364E35">
```
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Client/App.config]: You can remove that :+1:

File: samples/Nancy/Nancy.Client/App.config [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+  
+  <system.diagnostics>
+    <trace autoflush="true" />
+
+    <switches>
+      <add name="Microsoft.Owin" value="Verbose" />
+    </switches>
+  </system.diagnostics>
+
+  <system.web.webPages.razor>
```
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Client/App.config]: That too :+1:

File: samples/Nancy/Nancy.Client/Startup.cs [hunk 12]
```diff
@@ -81,14 +81,10 @@ public class Startup {
                             var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
 
                             // Add the access token to the returned ClaimsIdentity to make it easier to retrieve.
-                            notification.AuthenticationTicket.Identity.AddClaim(new Claim(
-                                type: OpenIdConnectParameterNames.AccessToken,
-                                value: payload.Value<string>(OpenIdConnectParameterNames.AccessToken)));
+                            notification.AuthenticationTicket.Identity.AddClaim(new Claim(OpenIdConnectParameterNames.AccessToken, payload.Value<string>(OpenIdConnectParameterNames.AccessToken)));
```
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Client/Startup.cs]: Was this change necessary?

File: samples/Nancy/Nancy.Client/Startup.cs [hunk 13]
```diff
@@ -106,7 +102,10 @@ public class Startup {
                 }
             });
 
-            app.UseNancy(options => options.PerformPassThrough = context => context.Response.StatusCode == HttpStatusCode.NotFound);
+            app.UseNancy(options => {
+                options.Bootstrapper = new NancyBootstrapper();
```
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Client/Startup.cs]: :+1:

File: samples/Nancy/Nancy.Client/project.json [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+  "version": "1.0.0-*",
+  "description": "OpenID Connect extensions for Owin.Security.OpenIdConnect.Server.",
+  "authors": [ "Kévin Chalet" ],
+  "owners": [ "Kévin Chalet" ],
+
+  "projectUrl": "https://github.com/aspnet-contrib/AspNet.Security.OpenIdConnect.Server",
+  "licenseUrl": "http://www.apache.org/licenses/LICENSE-2.0.html",
+
+  "commands": {
+    "Nancy.Client": "Nancy.Client"
```
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Client/project.json]: Please consider naming the command `run`.

File: samples/Nancy/Nancy.Client/project.json [hunk 15]
```diff
@@ -0,0 +1,33 @@
+﻿{
+  "version": "1.0.0-*",
+  "description": "OpenID Connect extensions for Owin.Security.OpenIdConnect.Server.",
```
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Client/project.json]: Woooops :smile: 
You can remove the description, it's obviously useless.

You can also remove the authors and the owners properties :+1:

File: samples/Nancy/Nancy.Server/Modules/AuthorizationModule.cs (hunk 2 above)
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Server/Modules/AuthorizationModule.cs]: Sir, it's time to use C# 6 syntax, you DESERVED it! :smile: 

```csharp
OwinContext.Authentication.User?.Identity == null
```

File: samples/Nancy/Nancy.Server/Modules/AuthorizationModule.cs (hunk 3 above)
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Server/Modules/AuthorizationModule.cs]: Per our discussion on JabbR, you can remove the comment :+1:

File: samples/Nancy/Nancy.Server/Modules/AuthorizationModule.cs [hunk 16]
```diff
@@ -133,7 +134,7 @@ public class AuthorizationModule : NancyModule {
 
             Post["/connect/authorize/deny"] = parameters => {
                 this.RequiresMSOwinAuthentication();
-                this.ValidateCsrfToken();
+                //this.ValidateCsrfToken(); TODO Investigate failure and uncomment me
```
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Server/Modules/AuthorizationModule.cs]: Here too.

File: samples/Nancy/Nancy.Server/Modules/AuthorizationModule.cs [hunk 17]
```diff
@@ -188,7 +189,7 @@ public class AuthorizationModule : NancyModule {
             };
 
             Post["/connect/logout"] = parameters => {
-                this.ValidateCsrfToken();
+                //this.ValidateCsrfToken(); TODO Investigate failure and uncomment me
```
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Server/Modules/AuthorizationModule.cs]: And there.

File: samples/Nancy/Nancy.Server/Providers/AuthorizationProvider.cs [hunk 18]
```diff
@@ -27,8 +27,8 @@ public class AuthorizationProvider : OpenIdConnectServerProvider {
             using (var database = new ApplicationContext()) {
                 // Retrieve the application details corresponding to the requested client_id.
                 var application = await (from entity in database.Applications
-                                         where entity.ApplicationID == context.ClientId
-                                         select entity).SingleOrDefaultAsync(context.OwinContext.Request.CallCancelled);
+                    where entity.ApplicationID == context.ClientId
```
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Server/Providers/AuthorizationProvider.cs]: Wooops, can you revert that?

File: samples/Nancy/Nancy.Server/Providers/AuthorizationProvider.cs [hunk 19]
```diff
@@ -55,8 +55,8 @@ public class AuthorizationProvider : OpenIdConnectServerProvider {
...
```
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Server/Providers/AuthorizationProvider.cs]: And that?

File: samples/Nancy/Nancy.Server/Startup.cs [hunk 20]
```diff
@@ -23,8 +22,8 @@ public class Startup {
             app.UseWhen(context => context.Request.Path.StartsWithSegments(new PathString("/api")), map => {
                 map.UseJwtBearerAuthentication(new JwtBearerAuthenticationOptions {
                     AuthenticationMode = AuthenticationMode.Active,
-                    AllowedAudiences = new[] { "http://localhost:54541/" },
-                    IssuerSecurityTokenProviders = new[] { new X509CertificateSecurityTokenProvider("http://localhost:54541/", certificate) }
+                    AllowedAudiences = new[] {"http://localhost:54541/"},
```
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Server/Startup.cs]: And also these cosmetic changes?

File: samples/Nancy/Nancy.Server/project.json [hunk 21]
```diff
@@ -0,0 +1,42 @@
+﻿{
+  "version": "1.0.0-*",
+  "description": "OpenID Connect extensions for Owin.Security.OpenIdConnect.Server.",
```
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Server/project.json]: Same remark as for Nancy.Client (remove description, authors and owners)

File: samples/Nancy/Nancy.Server/project.json [hunk 22]
```diff
@@ -0,0 +1,42 @@
...
+  "authors": [ "Kévin Chalet" ],
+  "owners": [ "Kévin Chalet" ],
+
+  "projectUrl": "https://github.com/aspnet-contrib/AspNet.Security.OpenIdConnect.Server",
+  "licenseUrl": "http://www.apache.org/licenses/LICENSE-2.0.html",
+
+  "commands": {
+    "Nancy.Server": "Nancy.Server"
```
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Server/project.json]: `run`.

File: samples/Nancy/Nancy.Server/project.json [hunk 23]
```diff
@@ -0,0 +1,42 @@
...
+    "Microsoft.Owin.Security.Jwt": "3.0.1",
+    "Microsoft.Owin.Security.OAuth": "3.0.1",
+    "Microsoft.Owin.SelfHost": "3.0.1",
+    "Nancy.MSOwinSecurity": "1.0.1",
+    "Nancy.Owin": "1.3.0",
+    "Nancy.Viewengines.Razor": "1.3.0",
+    "Newtonsoft.Json": "6.0.8",
+    "NWebsec.Owin": "1.0.0",
+    "System.IdentityModel.Tokens.Jwt": "4.0.2.202250711",
+    "Owin.Security.OpenIdConnect.Server" :  { "version": "1.0.0-*", "target": "project" },
```
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Server/project.json]: Fun. Is this now needed when referencing a local project?

File: samples/Nancy/Nancy.Client/App.config (hunk 10 above)
Comment (ilmax) [inline samples/Nancy/Nancy.Client/App.config]: doesn't DefaultRazorConfiguration use that sections to read usings and references? see [this](https://github.com/NancyFx/Nancy/blob/master/src/Nancy.ViewEngines.Razor/DefaultRazorConfiguration.cs)

File: samples/Nancy/Nancy.Client/Startup.cs (hunk 12 above)
Comment (ilmax) [inline samples/Nancy/Nancy.Client/Startup.cs]: nope, will undo!

File: samples/Nancy/Nancy.Client/App.config (hunk 10 above)
Comment (PinpointTownes) [inline samples/Nancy/Nancy.Client/App.config]: In my local clone of your fork, it works fine without this stuff :smile:

File: samples/Nancy/Nancy.Server/Providers/AuthorizationProvider.cs (hunk 18 above)
Comment (ilmax) [inline samples/Nancy/Nancy.Server/Providers/AuthorizationProvider.cs]: sure

File: samples/Nancy/Nancy.Server/project.json (hunk 23 above)
Comment (ilmax) [inline samples/Nancy/Nancy.Server/project.json]: not needed, it's for ensuring not consuming a package

**Comment by PinpointTownes** [comment]
@ilmax thank you very much for your contribution! :clap: 

Merged: https://github.com/aspnet-contrib/AspNet.Security.OpenIdConnect.Server/commit/491e45351d44f78508a18d93110ba2887f1e6b20 :tada:
